internal/gatewayapi/runner/runner.go

```diff
@@ -56,15 +57,21 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 			// so when a delete is triggered, delete all IR keys
 			if update.Delete || val == nil {
 				r.deleteAllIRKeys()
+				r.deleteAllStatusKeys()
 				return
 			}
 
-			var curKeys, newKeys []string
+			// IR keys for watchable
+			var curIRKeys, newIRKeys []string
+
 			// Get current IR keys
 			for key := range r.InfraIR.LoadAll() {
-				curKeys = append(curKeys, key)
+				curIRKeys = append(curIRKeys, key)
 			}
 
+			// Get the current DeletableStatus which manages status keys to be deleted
```

Comment:
lets add a comment here explaining the logic
1. Lets get all status keys from watchable and save it in this structure
2. Lets delete any current keys that are valid from this structure
3. The remaining keys, are the keys that are no longer valid, and will be deleted before we exit this function